libraries/rhsm_register.rb

```diff
@@ -0,0 +1,78 @@
+#
+# Author:: Chef Partner Engineering (<partnereng@chef.io>)
+# Copyright:: Copyright (c) 2015 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module RhsmCookbook
+  class RhsmRegister < ChefCompat::Resource
+    include RhsmCookbook::RhsmHelpers
+
+    resource_name :rhsm_register
+
+    property :_name_unused,        kind_of: String, name_property: true
+    property :activation_key,      kind_of: [ String, Array ]
+    property :satellite_host,      kind_of: String
+    property :organization,        kind_of: String
+    property :environment,         kind_of: String
+    property :username,            kind_of: String
+    property :password,            kind_of: String
+    property :auto_attach,         kind_of: [ TrueClass, FalseClass ], default: false
+
+    action :register do
+      remote_file '/tmp/katello-package.rpm' do
+        source "http://#{satellite_host}/pub/katello-ca-consumer-latest.noarch.rpm"
```

Comment:
It's a good question, but according to the Red Hat Satellite user guide, it's not a configurable URL, and this URL should always point to the latest katello CA available. 

My vote is to leave it as-is for now, but if it comes up with a use case, I'll definitely change it once I understand it more.  Fair?